src/openalea/lpy/lsysparameters.py

```diff
@@ -0,0 +1,652 @@
+        scalar.category = category
+
+        self._add_scalar(category, scalar)
+
+    def _add_scalar(self, category, scalar):
+        categoryobj = self.get_category(category)
+        categoryobj.scalars.append(scalar)
+
+    def add_function(self, name, value, category = None):
+        self.add_graphicalparameter(name, value, 'Function', category)
+
+    def add_graphicalparameter(self, name, value, ptype = None, category = None):
+        assert ptype in self.get_available_graphical_types()
+        if ptype is None:
+            ptype = graphictypemap[type(value)]
+        if category is None:
+            category = self.default_category_name
+
+        manager = self.get_graphicalparameter_managers()[ptype]
+        manager.setName(value, name)
+
+        self._add_graphicalparameter(category, manager, value)
+
+    def _add_graphicalparameter(self, category, manager, value):
+        categoryobj = self.get_category(category)
+        categoryobj.items.append((manager, value))
+
+    def set_option(self, name, value):
+        self.execOptions[name] = value
+
+    def get_option(self, name):
+        self.execOptions[name]
+
+    def get_options(self):
+        return self.execOptions
+
+    def set_color(self, index, value, name = None):
+        assert isinstance(value, pgl.Appearance)
+        self.color_list[index] = value
+        if not name is None:
+            value.name = name
+        elif not value.isNamed():
+            value.name = 'Color_'+str(index)
+
+    def get_color(self, index):
+        return self.color_list.get(index)
+
+    def get_colors(self):
+        return self.color_list
+
+    def get_categories(self):
+        return self.categories
+
+    def get_category_names(self):
+        return self.categories.keys()
+
+    def get_category(self, name = None):
+        if name is None:
+            name = self.default_category_name
+        if not name in self.categories:
+            category = Category({ 'name' : name, 'enabled' : True })
+            self.categories[name] = category
+        return self.categories[name]
+
+    def get_category_info(self, name = None):
+        """ If no name are given, the default category info is returned """
+        return self.get_category(name).info
+
+    def get_category_parameters(self, name = None) :
+        """ If no name are given, the default category parameters are returned """
+        return self.get_category(name).items+self.get_category(name).scalars
+
+    def get_category_graphicalparameters(self, name = None):
+        """ If no name are given, the default category graphical parameters are returned """
+        return self.get_category(name).items
+
+    def get_category_scalars(self, name = None):
+        """ If no name are given, the default category scalars are returned """
+        return self.get_category(name).scalars
+
+    def get_scalars(self):
+        return sum([ category.scalars for category in self.categories],[])
```

Comment:
Here is a bug.
`AttributeError: 'str' object has no attribute 'scalars'`

Reply:
fixed in e69847f5a26508360fa97df8648eddafdc8c1c2d